mutable_tree.go

```diff
@@ -760,10 +761,8 @@ func (tree *MutableTree) saveFastNodeVersion() error {
 		return err
 	}
 
-	if !tree.ndb.isFastStorageEnabled() {
```

Comment:
This is removed because its only called if `tree.ndb.isFastStorageEnabled` is true?

If so, can we add a comment stating this above the function?

Reply:
The reason why this was removed is that now we are going with the assumption that fast cache storage will be upgraded to on all versions above [1.1.0 or fastStorageVersionValue](https://github.com/osmosis-labs/iavl/blob/c293938ac371b008a2ac2f115b11093b49eafa5a/nodedb.go#L31). Therefore, we must make sure that we persist the latest live version on disk. Otherwise, re-upgrade protection would not function correctly.

In case there is any confusion, there are now 2 separate methods that check for fast storage to be enabled:
- in nodeDB (`hasUpgradedToFastStorage`)
  * This method checks that the upgrade has happened. When the upgrade is successful, fast storage on disk must match the latest tree.
  * I renamed this method from `isFastStorageEnabled` to `hasUpgradedToFastStorage` and added a comment for clarity 
- in immutable tree (`isFastStorageEnabled`)
   * This one checks for 2 conditions:
      1. The tree is of the latest version
      2. hasUpgradedToFastStorage - this one is more of a sanity check since now we are assuming that the upgrade should persist forever unless manually downgraded.
   * also added a comment

Let me know if this makes sense